src/main/java/org/opensearch/knn/quantization/models/quantizationParams/QuantizationParams.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.models.quantizationParams;
+
+import java.io.Externalizable;
+
+/**
+ * Interface for quantization parameters.
+ * This interface defines the basic contract for all quantization parameter types.
+ * It provides methods to retrieve the quantization type and a unique type identifier.
+ * Implementations of this interface are expected to provide specific configurations
+ * for various quantization strategies.
+ */
+public interface QuantizationParams extends Externalizable {
```

Comment:
Should we implement OpenSearch Writeable interface for this instead of Externalizable? https://github.com/opensearch-project/OpenSearch/blob/main/libs/core/src/main/java/org/opensearch/core/common/io/stream/Writeable.java

Reply:
Idea is isloate framewoek as much as isloated with intefaces , and keep interfces clean to remove tight coupling ,  Externalizable is also same as writable which inturns uses objectstream.

Reply:
I took depper look in Writable , seems promising though we have coupled with OpenSearch Core , Implemented using Writable.